### DISCUSSION TITLE
AROS ROM version

### DISCUSSION THREAD

**Discussion by wepl** [vAmigaWeb/vAmigaWeb#216] [Q&A]
The AROS rom version used is from 2019. I wonder if there could be build/used a newer one which maybe is more compatible?

**Comment by mithrendal** [comment]
Sure if there is a more compatible build lets use that instead of the 2019 build. You can swap kickstart roms in the settings and try out other AROS roms. If there is a better one we can use that instead. Which one is used in Winuae?

**Comment by wepl** [comment]
The actual AROS roms from snapshots https://sourceforge.net/projects/aros/files/nightly2/20240313/Binaries/AROS-20240313-amiga-m68k-boot-iso.zip/download are working only with OCS Agnus. With ECS Agnus there are gfx corruptions. Could be a bug in AROS or vAmiga. As OCS Agnus only allows 512 K Chip this limits the usability.
On the other side this newer AROS rom fixes a problem that color00 is not restored when returning from WHDLoad.

**Comment by dirkwhoffmann** [comment]
Since I cannot run AROS on my real Amigas (I don't want to burn Roms), I've run it in FSUAE for comparison (I can't run UAE, because I don't have Windows on the Apple M1):

A500 setting: 
<img width="1072" alt="Bildschirmfoto 2024-03-13 um 15 06 02" src="https://github.com/vAmigaWeb/vAmigaWeb/assets/12561945/36df627e-5f44-4f1e-82f9-4d362eb4bbb7">

There are graphics glitches at the bottom.

A500+ setting:

<img width="1072" alt="Bildschirmfoto 2024-03-13 um 15 04 41" src="https://github.com/vAmigaWeb/vAmigaWeb/assets/12561945/7a1970d7-12fd-4a89-bc21-f03a686a9096">

It didn't work at all.

vAmiga ECS Agnus / OCS Denise:

<img width="872" alt="Bildschirmfoto 2024-03-13 um 15 01 26" src="https://github.com/vAmigaWeb/vAmigaWeb/assets/12561945/d09c546b-cc8c-4fe0-aac3-08ab4b8e32de">

Graphics glitches at the top.

vAmiga ECS Agnus / ECS Denise:

<img width="872" alt="Bildschirmfoto 2024-03-13 um 15 12 41" src="https://github.com/vAmigaWeb/vAmigaWeb/assets/12561945/817b996d-1f24-4bf5-ba1a-54ba3495f37f">

Title screen looks good.

To be honest, I am not a big fan of AROS because, in the past, I've encountered many releases that didn't seem to work at all (at least not in UAE). Furthermore, only a few titles are compatible, limiting its use to a simple bootloader for programs that take over the entire machine after loading. In addition, there seems to be no release page with official (stable) releases. I've only found the repo with the nightly builds.

**Reply by wepl** [reply]
I agree, but it's the only way to avoid the licence concerns. To run WHDLoad it is sufficient.

**Reply by wepl** [reply]
The title screen doesn't matter to me. If booted with a disk some of the screen pointer seem to roll. 
<img width="679" alt="Bildschirmfoto 2024-03-13 um 22 04 00" src="https://github.com/vAmigaWeb/vAmigaWeb/assets/3184445/af08f3f6-d8df-4cb9-9727-b911dfd84bbe">
Looks fine with OCS Agnus.
If we would know it's an AROS bug I could file a report there.

**Reply by wepl** [reply]
Testet with amiga-m68k-boot-iso Nightly Build 2024-03-13 http://www.aros.org/de/nightly1.php

**Comment by mithrendal** [comment]
https://github.com/dirkwhoffmann/vAmiga/issues/899

**Comment by mithrendal** [comment]
Hi @wepl 

@dirkwhoffmann found the culprit of the glitch in vAmigas ECS Agnus implementation and he fixed it yesterday. 

vAmigaWeb 4.1 has been just published which contains the fix.

see also for reference 
https://github.com/dirkwhoffmann/vAmiga/issues/899


so you can now configure also ECS Denise on whdload... 😎
